### PR TITLE
Ben - L101.nonghg_en_USA_S_T_Y

### PR DESCRIPTION
**Pull request by bpbond**
Code translated. Note a few comments and input headers need to be filled in/clarified (see #255 ).

### CODE REVIEW

**Comment by bpbond** [comment]
@rynacui - do you mind doing a code review on this? Thanks.

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      mutate(technology = fuel,
+             fuel = paste(mode, size.class, sep = "_"),
+             sector = UCD_sector) %>%
+      select(-mode, -UCD_sector, -size.class, -UCD_technology, -UCD_fuel) %>%
+
+      # Bind all together
+      bind_rows(L1231.in_EJ_R_elec_F_tech_Yh, temp) ->
+      L101.in_EJ_R_en_Si_F_Yh
+
+    L101.in_EJ_R_en_Si_F_Yh %>%
```
Comment (rynacui) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Are these temp-data with `Xyear` & wide format going to be replaced by outputs with `year` & long format later? And these lines will be taken out?

Comment (bpbond) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Maybe, but this is a little tricky. It turns out that there are multiple lines for each combination of year and sector, fuel, region, and technology, which means that once we `gather` it, the dataset can't be re-`spread` (an error is thrown if we try). So we may need to pass these data in wide format, although definitely without the Xyears.

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+    # Compute SO2, CO, NOx, VOC, and NH3 emissions factors by dividing EPA inventory by IEA energy balances
+    # We do this for each gas, so define a function with all the steps
+    EPA_compute_emissions_factors <- function(x, L101.in_EJ_USA_en_Sepa_F_Yh.mlt) {
+      L101.in_EJ_USA_en_Sepa_F_Yh.mlt %>%
+        left_join(x, by = c("EPA_agg_sector" = "sector", "EPA_agg_fuel" = "fuel", "year" = "year")) %>%
+        rename(emissions = value) %>%
+        mutate(emissions = if_else(is.na(emissions), 0, emissions),
+               em_factor = emissions / energy,
+               em_factor = if_else(is.nan(em_factor) | is.infinite(em_factor), 0, em_factor)) ->
```
Comment (rynacui) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Why `is.nan` here instead of `is.na`? Are they different?

Comment (bpbond) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Yes, `NaN` means "not a number", for example `sqrt(-1)`. This logic follows the original code, except replacing a string comparison (`em_factor == "NaN"`) with `is.nan`.

Comment (rynacui) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Got it. Thanks!

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        x_em_factor
+
+      # Compute sector emissions. If these are zero, then reset all fuel em_factors to 1
```
Comment (rynacui) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Just out of curiosity, when sector emissions equal zero, why em_factors are set to 1 Tg/EJ?

Comment (bpbond) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Good question; the documentation should make this clear. @pkyle could you weigh in?

**Comment by bpbond** [comment]
Thanks Ryna!

**Comment by cahartin** [comment]
@bpbond and @rynacui The emissions inputs need to be updated with title and source.

**Comment by bpbond** [comment]
@cahartin right, thanks. I have queries out to @pkyle in #255

**Comment by cahartin** [comment]
@bpbond gotcha. thanks.

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R (hunk 3 above)
Comment (pkyle) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: I looked at the code but also couldn't tell; @kvcalvin  can you comment?

Comment (bpbond) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: @kvcalvin - when you get a chance

Comment (kvcalvin) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: Good question. I don't totally remember, but I think there was a specific case where this was required. Essentially, we will later scale these emissions to match EDGAR totals. If there is a zero here, but positive emissions in EDGAR, then the scaling won't work.

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R (hunk 1 above)
Comment (d3y419) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: see also: https://github.com/JGCRI/gcamdata/pull/309

File: R/zchunk_L101.nonghg_en_USA_S_T_Y.R (hunk 3 above)
Comment (d3y419) [inline R/zchunk_L101.nonghg_en_USA_S_T_Y.R]: https://github.com/JGCRI/gcamdata/issues/327

**Pull request merged**